### PR TITLE
Support "Unlimited" when editing BSP cap

### PR DESCRIPTION
**Pull request by EndymionJkb**
Changes proposed in this pull request:
- Support "Unlimited" when editing BSP cap
- Ensure proper type on etherscanLink calls
- Read controller from contract vs graph, to support setController

### CODE REVIEW

**Review by bonustrack — COMMENTED** [review]
Added few comments, can you change the base branch from "smartpool" to "develop"? (develop is the new smartpool :))

File: src/components/Modal/EditCap.vue [hunk 1]
```diff
@@ -38,19 +47,24 @@
...
 export default {
   props: ['open', 'value', 'pool'],
   data() {
     return {
       loading: false,
-      input: false
+      input: false,
+      type: false,
+      capInputOptions,
+      MAX
```
[the remark this answers is not in the document]
Comment (EndymionJkb) [reply src/components/Modal/EditCap.vue]: Added with last merge/commit

File: src/helpers/utils.ts [hunk 2]
```diff
@@ -11,6 +11,8 @@ export const MAX_GAS = new BigNumber('0xffffffff');
 export const MAX_UINT = MaxUint256;
 export const POOL_TOKENS_DECIMALS = 18;
 export const GAS_LIMIT_BUFFER = 0.1;
+export const MAX =
+  '115792089237316195423570985008687907853269984665640564039457.584007913129639935';
```
Comment (bonustrack) [inline src/helpers/utils.ts:15]: How did you get this number?

**Review by bonustrack — APPROVED** [review]

File: src/helpers/utils.ts (hunk 2 above)
Comment (EndymionJkb) [reply src/helpers/utils.ts:15]: Ultimately, defined in the golang backend
#1 - eth/api_backend.go 
math.MaxBig256 == 115792089237316195423570985008687907853269984665640564039455084007913129639935

It's in the tests, too (Javascript number limit). I'm not super comfortable with needing these magic numbers, but it's just how Javascript and Ethereum libraries seem to have to work together

Comment (EndymionJkb) [reply src/helpers/utils.ts:15]: It's worse in the tests:
    const MAX = web3.utils.toTwosComplement(-1);
    const MaxBig256 = '115792089237316195423570985008687907853269984665640564039457.584007913129639935';

You have to pass in MAX, and read MaxBig256. In Vue you seem to be able to read and write the same MaxBig256 value

**Pull request merged**